imagetest/test_suites/metadata/setup.go

```diff
@@ -56,63 +205,54 @@ func TestSetup(t *imagetest.TestWorkflow) error {
 	if err != nil {
 		return err
 	}
-	vm2.SetShutdownScript(shutdownScript)
+	for _, ms := range metadataScripts {
+		if ms.url {
+			vm2.SetMetadataScriptURL(ms.metadataKey, ms.script(), ms.extension)
+		} else {
+			vm2.AddMetadata(ms.metadataKey, ms.script())
+		}
```

Comment:
if you're already putting logic in method form you could just put this logic in the script() method (and omit extension param as mentioned above)

Reply:
OK, I've looked at this for a bit and I don't see a way to put this logic into the script() method. AddMetadata and SetMatadataScriptURL do very different things, even though the values passed to them are similar.